pkg/k8sutils/job.go

```diff
@@ -127,12 +126,26 @@ func (k8s *k8sImpl) CreateK8sJob(
 							},
 							Env: []v1.EnvVar{
 								{
-									Name:  "CONFIGURATION_SERVICE",
-									Value: jobSettings.InitContainerConfigurationServiceAPIEndpoint,
+									Name: "CONFIGURATION_SERVICE",
+									ValueFrom: &v1.EnvVarSource{
+										ConfigMapKeyRef: &v1.ConfigMapKeySelector{
+											LocalObjectReference: v1.LocalObjectReference{
+												Name: "job-service-config",
+											},
+											Key: "init_container_configuration_endpoint",
+										},
+									},
 								},
 								{
-									Name:  "KEPTN_API_TOKEN",
-									Value: jobSettings.KeptnAPIToken,
+									Name: "KEPTN_API_TOKEN",
+									ValueFrom: &v1.EnvVarSource{
+										SecretKeyRef: &v1.SecretKeySelector{
+											LocalObjectReference: v1.LocalObjectReference{
+												Name: "job-serivce-keptn-secrets",
+											},
+											Key: "token",
+										},
+									},
```

Comment:
Same problem with the secret. I am in favour of not having the secret directly in the manifest though.

Reply:
I tested it with jobs that had the following service account (https://github.com/keptn-contrib/job-executor-service/blob/4e9505a03c5a53c3349a9ce5e75a432a0f2b2265/chart/templates/jobserviceaccount.yaml) and it worked. So I'm not exactly sure how the permissions are for these references.

Reply:
No additional permissions are necessary for the job pod, because we just reference the secret and aren't asking over the API endpoint